integration-tests/maven/src/test/java/io/quarkus/maven/it/SurefireRerunFailedTestsIT.java

```diff
@@ -0,0 +1,64 @@
+package io.quarkus.maven.it;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.maven.shared.invoker.MavenInvocationException;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.maven.it.verifier.MavenProcessInvocationResult;
+import io.quarkus.maven.it.verifier.RunningInvoker;
+
+/**
+ * Test that mvn -Dsurefire.rerunFailingTestsCount=2 test works with QuarkusTests.
+ */
+@DisableForNative
+public class SurefireRerunFailedTestsIT extends RunAndCheckMojoTestBase {
+
+    @Test
+    public void testFlakyTestsInSingleProfileCase() throws MavenInvocationException, IOException {
+        testDir = initProject("projects/test-flaky-test-single-profile", "projects/test-flaky-test-single-profile-processed");
+        final RunningInvoker running = new RunningInvoker(testDir, false);
+
+        // trigger mvn verify -Dsurefire.rerunFailingTestsCount=2
+        final String[] mvnArgs = new String[] { "verify", "-Dsurefire.rerunFailingTestsCount=3" };
```

Comment:
Should the comment be aligned with the actual argument value passed?

Reply:
I'll update the comment to something generic so that we don't need to double-maintain if we make changes in the test.